main.go

```diff
@@ -130,93 +121,36 @@ func main() {
 
 	cacherClient, err := client.New(*facility)
 	if err != nil {
-		panic(err)
+		logger.Fatal(err, "Failed to create the cacher client")
 	}
-	go func() {
-		c := time.Tick(15 * time.Second)
-		for range c {
-			// Get All hardware as a proxy for a healthcheck
-			// TODO (patrickdevivo) until Cacher gets a proper healthcheck RPC
-			// a la https://github.com/grpc/grpc/blob/master/doc/health-checking.md
-			// this will have to do.
-			// Note that we don't do anything with the stream (we don't read from it)
-			var isCacherAvailableTemp bool
-			ctx, cancel := context.WithCancel(context.Background())
-			_, err := cacherClient.All(ctx, &cacher.Empty{})
-			if err == nil {
-				isCacherAvailableTemp = true
-			}
-			cancel()
-
-			isCacherAvailableMu.Lock()
-			isCacherAvailable = isCacherAvailableTemp
-			isCacherAvailableMu.Unlock()
-
-			logger.With("status", isCacherAvailableTemp).Debug("tick")
-		}
-
-	}()
 
 	grpcServer := grpc.NewServer(serverOpts...)
-	hegel.RegisterHegelServer(grpcServer, &server{
+	hegelServer = &server{
 		log:          logger,
 		cacherClient: cacherClient,
-	})
+	}
 
-	// Register grpc prometheus server
+	hegel.RegisterHegelServer(grpcServer, hegelServer)
 	grpc_prometheus.Register(grpcServer)
-	http.Handle("/metrics", promhttp.Handler())
-	http.HandleFunc("/_packet/healthcheck", healthCheckHandler)
-	http.HandleFunc("/_packet/version", versionHandler)
-	logger.With("port", *metricsPort).Info("Starting http server")
-	go func() {
-		err := http.ListenAndServe(fmt.Sprintf(":%d", *metricsPort), nil)
-		if err != nil {
-			logger.Error(err, "failed to serve http")
-			panic(err)
-		}
-	}()
```

Comment:
Why was this changed around?

Reply:
Because I moved all the http calls in a new file called http_handler.go